frontend/src/Components/HomePage/HomePage.jsx

```diff
@@ -1,17 +1,26 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import MainHeader from "../MainHeader/MainHeader";
 import "./HomePage.css";
 import BusinessList from "../BusinessList/BusinessList";
 
 const HomePage = () => {
   const [selectedCategory, setSelectedCategory] = useState("");
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    const storedUser = JSON.parse(localStorage.getItem("user"));
+    if (storedUser) {
+      setUser(storedUser);
+    }
```

Comment:
the call to JSON.parse could potentially throw an error, so it might be good to wrap this in a try-catch. Alternatively, you could wrap the component in an Error Boundary, but I'm guessing the course hasn't covered this yet.

Reply:
Done 👍🏼. By the way, we are actually done with classes and are left to do independent learning now.